neo4j-jdbc/src/main/java/org/neo4j/jdbc/ConnectionImpl.java

```diff
@@ -661,6 +691,42 @@ public void flushTranslationCache() {
 		}
 	}
 
+	private <T extends AutoCloseable> T addAutoCloseable(T autoCloseable) {
```

Comment:
Naming: "regiterStatement" or similar. Plus I would hook up the this connection as listener on the statement here and only here. So you have the machinery decoupled.

Reply:
Updated as agreed.